src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -406,6 +412,10 @@ def process_parquet_queue_with_threads(
         while len(completed_partitions) < total_partitions or thread_futures:
             # Process any completed futures and handle errors
             process_completed_futures(thread_futures)
+            while not process_or_thread_error_indicator.empty():
+                gracefully_exited_processes.add(
+                    process_or_thread_error_indicator.get(block=False)[0]
+                )
```

Comment:
if we use the same queue, we don't need the while here, we can handle in the following `elif parquet_id = ...` logic